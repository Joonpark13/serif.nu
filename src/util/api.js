/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
