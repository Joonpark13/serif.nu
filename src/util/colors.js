export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
