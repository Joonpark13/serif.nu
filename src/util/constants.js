/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
