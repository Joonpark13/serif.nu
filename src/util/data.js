import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
