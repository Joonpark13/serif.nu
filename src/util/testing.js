/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
