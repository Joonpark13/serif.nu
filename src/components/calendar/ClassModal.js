import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import { getFormattedClassSchedule } from 'util/time';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Divider from '@material-ui/core/Divider';

export const styles = {
  dialog: {
    top: '12.5%',
    left: '12.5%',
    width: '500px',
    padding: '10px',
  },
};

function ClassModal({ showDialog, toggleDialog, classes, section, removeSection }) {
  return (
    <Dialog
      open={showDialog}
      onClose={toggleDialog}
      aria-labelledby="simple"
      aria-describedby="simple-dialog-description"
    >
      <div className={classes.dialog}>
        <Typography gutterBottom variant="h5">
          {`${section.subjectId} ${section.courseId} Section ${section.sectionNumber}`}
        </Typography>
        <Divider />
        <br />

        {section.schedules.map(schedule => (
          <div key={JSON.stringify(schedule)}>
            <Typography>{getFormattedClassSchedule(schedule)}</Typography>
            <Typography>{schedule.location}</Typography>
          </div>
        ))}
        {section.instructors.map(
          instructor => <Typography key={instructor}>{instructor}</Typography>,
        )}
        <br />

        {section.descriptions.map(description => (
          <div key={JSON.stringify(description)}>
            <Typography variant="h6">{description.name}</Typography>
            <Typography gutterBottom>{description.value}</Typography>
            <br />
          </div>
        ))}
      </div>
      <DialogActions>
        <Button onClick={() => removeSection(section.id, section.color)}>Remove</Button>
        <Button onClick={toggleDialog}>Cancel</Button>
      </DialogActions>
    </Dialog>
  );
}

ClassModal.propTypes = {
  section: PropTypes.objectOf(PropTypes.any).isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  showDialog: PropTypes.bool.isRequired,
  toggleDialog: PropTypes.func.isRequired,
  removeSection: PropTypes.func.isRequired,
};

export { ClassModal as UnstyledClassModal };
export default withStyles(styles)(ClassModal);
