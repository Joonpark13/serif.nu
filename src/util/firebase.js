import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
