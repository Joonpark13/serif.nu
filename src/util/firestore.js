import firebase from 'firebase/app';
import 'firebase/firestore';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export default firebase.firestore();
