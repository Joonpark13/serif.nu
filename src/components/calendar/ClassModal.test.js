import Button from '@material-ui/core/Button';
import { wrapperCreator } from 'util/testing';
import { UnstyledClassModal, styles } from './ClassModal';

describe('ClassModal', () => {
  const section = {
    id: '198732',
    sectionNumber: 20,
    topic: 'Section topic...',
    schedules: [{
      location: 'somewhere',
      dow: ['Mo'],
      start: {
        hour: 10,
        minute: 30,
      },
      end: {
        hour: 12,
        minute: 0,
      },
    }],
    instructors: ['A prof'],
    color: '#58B947',
    descriptions: [{ name: '', value: '' }],
  };
  const defaultProps = {
    section,
    showDialog: true,
    toggleDialog: () => {},
    removeSection: () => {},
  };

  const getWrapper = wrapperCreator(UnstyledClassModal, defaultProps, styles);

  it('renders correctly', () => {
    const wrapper = getWrapper();

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('closes the modal when clicked', () => {
    const removeSectionMock = jest.fn();
    const wrapper = getWrapper({ removeSection: removeSectionMock });
    wrapper.find(Button).first().simulate('click');

    expect(removeSectionMock).toHaveBeenCalledWith(section.id, section.color);
  });
});
