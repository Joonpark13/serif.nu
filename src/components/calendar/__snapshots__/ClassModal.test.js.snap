// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassModal renders correctly 1`] = `
<WithStyles(Dialog)
  aria-describedby="simple-dialog-description"
  aria-labelledby="simple"
  onClose={[Function]}
  open={true}
>
  <div
    className=""
  >
    <WithStyles(Typography)
      gutterBottom={true}
      variant="h5"
    >
      undefined undefined Section 20
    </WithStyles(Typography)>
    <WithStyles(Divider) />
    <br />
    <div>
      <WithStyles(Typography)>
        Mo 10:30 AM - 12:00 PM
      </WithStyles(Typography)>
      <WithStyles(Typography)>
        somewhere
      </WithStyles(Typography)>
    </div>
    <WithStyles(Typography)>
      A prof
    </WithStyles(Typography)>
    <br />
    <div>
      <WithStyles(Typography)
        variant="h6"
      >
        
      </WithStyles(Typography)>
      <WithStyles(Typography)
        gutterBottom={true}
      >
        
      </WithStyles(Typography)>
      <br />
    </div>
  </div>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      onClick={[Function]}
    >
      Remove
    </WithStyles(Button)>
    <WithStyles(Button)
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
